src/crypto/crypto.js

```diff
@@ -94,35 +81,32 @@ export async function publicKeyEncrypt(algo, publicParams, data, fingerprint) {
  * @param {module:enums.publicKey}        algo              Public key algorithm
  * @param {Object}                        publicKeyParams   Algorithm-specific public key parameters
  * @param {Object}                        privateKeyParams  Algorithm-specific private key parameters
- * @param {Array<module:type/mpi|
-             module:type/ecdh_symkey>}    data_params       Encrypted session key parameters
+ * @param {Object}                        sessionKeyParams  Encrypted session key parameters
  * @param {Uint8Array}                    fingerprint       Recipient fingerprint
  * @returns {Uint8Array}                                    Decrypted data
  * @async
  */
-export async function publicKeyDecrypt(algo, publicKeyParams, privateKeyParams, data_params, fingerprint) {
+export async function publicKeyDecrypt(algo, publicKeyParams, privateKeyParams, sessionKeyParams, fingerprint) {
   switch (algo) {
     case enums.publicKey.rsaEncryptSign:
     case enums.publicKey.rsaEncrypt: {
-      const c = data_params[0].toUint8Array();
+      const { c } = sessionKeyParams;
       const { n, e } = publicKeyParams;
       const { d, p, q, u } = privateKeyParams;
       return publicKey.rsa.decrypt(c, n, e, d, p, q, u);
     }
     case enums.publicKey.elgamal: {
-      const c1 = data_params[0].toUint8Array();
-      const c2 = data_params[1].toUint8Array();
+      const { c1,c2 } = sessionKeyParams;
       const p = publicKeyParams.p;
       const x = privateKeyParams.x;
       return publicKey.elgamal.decrypt(c1, c2, p, x);
     }
     case enums.publicKey.ecdh: {
       const { oid, Q, kdfParams } = publicKeyParams;
       const { d } = privateKeyParams;
-      const V = data_params[0].toUint8Array();
-      const C = data_params[1].data;
+      const { V, C } = sessionKeyParams;
```

Comment:
to make this very confusing, these two lines could be:
```
      const { V, C: { data: C } } = sessionKeyParams;
      return publicKey.elliptic.ecdh.decrypt(
        oid, kdfParams, V, C.data, Q, d, fingerprint);
```

Reply:
I like it better as it is. Alternatively, you could store it as a `Uint8Array` directly, and get rid of the `ECDHSymkey` type entirely.

Reply:
ECDHSymkey needs a prefix of the length and the type abstracts that away with reading and writing.

```
write() {
    return util.concatUint8Array([new Uint8Array([this.data.length]), this.data]);
  }
```

I could parse the length in this function and add a special case to `serializeKeyParams` for ECDH. It currently treats all Uint8Array as MPI and uses the type's write method otherwise.

```
orderedParams = Object.keys(params).map(name => {
        const param = params[name];
        return util.isUint8Array(param) ? util.uint8ArrayToMpi(param) : param.write();
      });
```